### PR TITLE
update DVR Window Range for livestreams with infinite window size

### PR DESCRIPTION
**Pull request by IcoDeveloper**
Workaround for DVRWindow Range not updating on live MSS stream with infinite DVR window size #232

### CODE REVIEW

File: app/js/mss/MssFragmentController.js [hunk 1]
```diff
@@ -100,20 +100,22 @@ Mss.dependencies.MssFragmentController = function() {
                 return;
             }
             // In case of live streams, update segment timeline according to DVR window
-            else if (manifest.timeShiftBufferDepth && manifest.timeShiftBufferDepth > 0) {
+            else if ((manifest.timeShiftBufferDepth && manifest.timeShiftBufferDepth > 0) || manifest.canSeek ) {
```
[the remark this answers is not in the document]
Comment (IcoDeveloper) [reply app/js/mss/MssFragmentController.js]: Going for manifest.canseek only would be cleaner, yes.
However, we can't rely on the canseek property actually being available in the manifest:
`<SmoothStreamingMedia MajorVersion="2" MinorVersion="2" TimeScale="10000000" IsLive="TRUE" Duration="0" LookAheadFragmentCount="2" DVRWindowLength="300000000">` (from Azure Media Live Stream)
we'd have to set the canSeek property in the parser based on DVRWindowLength > 0 if the property is missing in XML manifest.

[the remark this answers is not in the document]
Comment (bbert) [reply app/js/mss/MssFragmentController.js]: Yes, ok for considering missing `CanSeek `property when `DVRWindowLength `is > 0.
Now we have to check if we can set `DVRWindowLength `to Infinity by default if not set & `CanSeek=TRUE`

**Comment by IcoDeveloper** [comment]
Thanks for the pointer regarding `DVRWindowLength = Infinity`. Makes for a less intrusive change.

**Comment by bbert** [comment]
Hi @IcoDeveloper , can you update the PR so that it integrates the changes as we discussed?

**Comment by bbert** [comment]
@IcoDeveloper , the Pr seems to be ok in fact. I have updated to clean the code and tested on my side.
Can you test on your side before I do merge?

**Pull request merged**